### PR TITLE
Add in a few developer docs

### LINKED ISSUE #1793: Unable to build from master branch
Based on https://github.com/coursier/coursier/blob/master/doc/DEVELOPMENT.md  I did

```sh
git clone --recursive https://github.com/coursier/coursier.git
sbt cli/pack
```

and got this error

```
❯ sbt cli/pack
[info] welcome to sbt 1.3.12 (N/A Java 14.0.1)
[info] loading settings for project coursier-build from plugins.sbt ...
[info] loading project definition from /Users/bobelev/Developer/src/coursier/project
/Users/bobelev/Developer/src/coursier/build.sbt:481: error: not found: value sbtCoursierVersion
        "PLUGIN_VERSION" -> sbtCoursierVersion,
                            ^
/Users/bobelev/Developer/src/coursier/build.sbt:482: error: not found: value sbtCoursierVersion
        "PLUGIN_EXTRA_SBT" -> extraSbt(sbtCoursierVersion),
                                       ^
```

### PR DESCRIPTION
**Pull request by ckipp01**
This just adds in a few notes in the contributing section. It's not a ton of information, but might be helpful for people looking to get started on the code base.

This closes #1793

### CODE REVIEW

File: doc/docs/dev-cli.md [hunk 1]
```diff
@@ -13,11 +13,15 @@ final `coursier` launcher generated upon each release relies on the
 `coursier bootstrap` command, itself pulling a proguarded assembly of the
 cli module of coursier.
 
+Also note that Cousier relies on a slightly different sbt than the default sbt
```
Comment (alexarchambault) [inline doc/docs/dev-cli.md:16]: Typo: Cousier -> Coursier :)

**Review by alexarchambault — COMMENTED** [review]
Thanks @ckipp01! Apart from the minor comment I added, LGTM

File: doc/docs/dev-cli.md (hunk 1 above)
Comment (ckipp01) [reply doc/docs/dev-cli.md:16]: ah jeez. ha, good catch.

Comment (ckipp01) [reply doc/docs/dev-cli.md]: Fixed.

**Review by alexarchambault — APPROVED** [review]
Thanks!

**Pull request merged**